### PR TITLE
pkg/libcose: bump version

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This updates pkg/libcose to current master.
The API evolved a bit, thus the test needs adoption.
Also, libcose now uses nanocbor.

~~Marking WIP until the test works again.~~ (tests updated)

### Testing procedure

- let CI confirm the test still works

### CODE REVIEW

**Comment by kaspar030** [comment]
This is not WIP anymore. Commit history got messy, so I rebased and squashed. Ready for review!

File: pkg/libcose/Makefile.dep [hunk 1]
```diff
@@ -8,3 +8,9 @@ endif
 ifneq (,$(filter libcose_crypt_hacl,$(USEMODULE)))
   USEPKG += hacl
 endif
+ifneq (,$(filter libcose_crypt_monocypher,$(USEMODULE)))
```
Comment (fjmolinas) [inline pkg/libcose/Makefile.dep]: @bergzand are `monocypher` and `c25519` still missing RIOT glue code? If its the case I might take them out of this PR for now.

Comment (fjmolinas) [reply pkg/libcose/Makefile.dep]: @kaspar030 testes #11818 using `moncypher` and `c25519` and they are not working yet.

I order not to delay I would suggest removing them from the PR, and going forward with the rest.

Comment (kaspar030) [reply pkg/libcose/Makefile.dep]: +1, I've removed the corresponding commit

**Comment by kaspar030** [comment]
&go! Thanks a lot for fixing up the tests @fjmolinas!

**Pull request merged**